File: bench/string_ops/string_ops.cpp
#include <iostream>
#include <string>
#include <algorithm>

// 10000回の upper-case + 結合
int main() {
    std::string result;
    for (int i = 0; i < 10000; i++) {
        std::string s = "item-" + std::to_string(i);
        std::transform(s.begin(), s.end(), s.begin(), ::toupper);
        result += s;
    }
    std::cout << result.length() << std::endl;
    return 0;
}

File: bench/string_ops/string_ops.zig
const std = @import("std");

// 10000回の upper-case + 結合
pub fn main() !void {
    var gpa = std.heap.GeneralPurposeAllocator(.{}){};
    defer _ = gpa.deinit();
    const allocator = gpa.allocator();

    var result = std.ArrayList(u8).empty;
    defer result.deinit(allocator);

    var i: usize = 0;
    while (i < 10000) : (i += 1) {
        var buf: [20]u8 = undefined;
        const s = std.fmt.bufPrint(&buf, "item-{d}", .{i}) catch unreachable;
        for (s) |c| {
            try result.append(allocator, std.ascii.toUpper(c));
        }
    }

    var out_buf: [4096]u8 = undefined;
    var writer = std.fs.File.stdout().writer(&out_buf);
    const stdout = &writer.interface;
    try stdout.print("{d}\n", .{result.items.len});
    try stdout.flush();
}
